### PR TITLE
Fabl rich text formatter

### PR DESCRIPTION
**Pull request by pjlee11**
THIS IS A PROOF OF CONCEPT

**Overall change:** Uses a local copy of the FABL rich text formatter and preprocessing the CPS candyXML.

### CODE REVIEW

File: data/indonesia/mediaAssetPage/media-49591990.json [hunk 1]
```diff
@@ -151,12 +151,22 @@
...
+                "markupType": "candy_xml",
+                "type": "paragraph"
+            },
             {
                 "text": "Salah satu peristiwa yang menjadi tajuk utama pemberitaan adalah saat Ahed Tamimi (16 tahun) dipenjara selama delapan bulan pada Desember 2017 karena didakwa memukul tentara Israel. ",
                 "markupType": "plain_text",
                 "type": "paragraph"
             },
-            {
+            {
```
Comment (pjlee11) [inline data/indonesia/mediaAssetPage/media-49591990.json]:
```suggestion
            {
```

**Comment by pjlee11** [comment]
Screenshot as of the commit `d6ede1512b3b5589952a41e4036dc718903311fa`

image

**Comment by dr3** [comment]
As of 4f14622b4552da3fb37cd79034e507624e5d3a0c

image

**Comment by dr3** [comment]
After 2fb183574f41b9c8c350d808fbf2a330111bb640 
image

File: src/app/lib/utilities/preprocessor/rules/cpsAsset/addHeadlineBlock/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import pathOr from 'ramda/src/pathOr';
+import path from 'ramda/src/path';
+
+const addHeadingToSTY = jsonRaw => {
+  const headlineText = pathOr(
```
Comment (ryanmccombe) [inline src/app/lib/utilities/preprocessor/rules/cpsAsset/addHeadlineBlock/index.js:5]: Just `path` would work the same

File: src/app/lib/utilities/preprocessor/rules/cpsAsset/convertCpsBlocksToOptimoBlocks/index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import { clone, pathOr } from 'ramda';
+import image from './generators/image';
+import {
+  candyXmlToRichText,
+  stringToRichText,
+} from '../../../../../../../../../fabl-modules/libraries/rich-text-transforms';
+
+const wrapText = innerXML => `<body><paragraph>${innerXML}</paragraph></body>`;
+
+const handleCandyXML = xml => candyXmlToRichText(xml);
```
Comment (ryanmccombe) [inline src/app/lib/utilities/preprocessor/rules/cpsAsset/convertCpsBlocksToOptimoBlocks/index.js:10]: Do we need this anonymous wrapper function that doesn't do anything?  Isn't `handleCandyXML` the same as `candyXmlToRichText`

File: src/app/lib/utilities/preprocessor/rules/cpsAsset/convertCpsBlocksToOptimoBlocks/index.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+const handlePlainText = block => stringToRichText(block.text);
+
+const handleMissingType = block => {
```
Comment (ryanmccombe) [inline src/app/lib/utilities/preprocessor/rules/cpsAsset/convertCpsBlocksToOptimoBlocks/index.js:14]: probably don't need to return null - console.log already returns undefined

File: src/app/lib/utilities/preprocessor/rules/cpsAsset/convertCpsBlocksToOptimoBlocks/index.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+  return parsedBlock;
+};
+
+const convertCpsBlocksToOptimoBlocks = jsonRaw => {
+  const json = clone(jsonRaw);
+  const blocks = pathOr([], ['content', 'blocks'], json);
+
+  const parsedBlocks = blocks
+    .map(block => {
```
Comment (ryanmccombe) [inline src/app/lib/utilities/preprocessor/rules/cpsAsset/convertCpsBlocksToOptimoBlocks/index.js:60]: Probably don't need the wrapper function - `map(parseBlockByMarkupType)`

File: src/app/routes/index.js [hunk 6]
```diff
@@ -38,7 +38,7 @@ const routes = [
   {
     path: mediaAssetPageRegexPath,
     exact: true,
-    component: MediaPage,
+    component: Article,
```
Comment (ryanmccombe) [inline src/app/routes/index.js:41]: mergeit

**Comment by pjlee11** [comment]
I'm going to split this out into a basic preprocessor rule just supporting the paragraph, so that it's easier to review and more feasible to merge

**Pull request closed without merging**